code/ReportAdmin.php

```diff
@@ -227,28 +239,60 @@ public function getEditForm($id = null, $fields = null)
             $fields = $report->getCMSFields();
         } else {
             // List all reports
-            $fields = new FieldList();
+            $fields = FieldList::create();
             $gridFieldConfig = GridFieldConfig::create()->addComponents(
+                // This is a container component that is required by filter header component
+                GridFieldButtonRow::create('before'),
+                $filterHeader = GridFieldFilterHeader::create(),
                 GridFieldSortableHeader::create(),
-                GridFieldDataColumns::create(),
-                GridFieldFooter::create()
+                $columns = GridFieldDataColumns::create(),
+                GridFieldPageCount::create(),
+                GridFieldPaginator::create()
             );
+
+            // Configure the filter header filter search form
+            $generalField = BasicSearchContext::config()->get('general_search_field_name');
+            $searchFieldList = FieldList::create([
+                HiddenField::create($generalField),
+                TextField::create('Title'),
+                TextField::create('Description'),
+            ]);
+
+            $searchContext = BasicSearchContext::create(Report::class);
+            $searchContext->setFields($searchFieldList);
+
+            // Setup filter configuration - partial match with case-insensitive modifier
+            $filters = [
+                'Title',
+                'Description',
+            ];
+
```

Comment:
```suggestion
            $searchContext = BasicSearchContext::create(Report::class);
            $searchContext->setFields($searchFieldList);

            // Setup filter configuration - partial match with case-insensitive modifier
            $filters = [
                'Title',
                'Description',
            ];
```
This is very subjective so I won't block merging if it's not done - but IMO this is slightly easier on the eyes. (just removing the first and last empty new lines here)